### PR TITLE
AdditiveSharingTensor

### PR DESCRIPTION
**Pull request by iamtrask**
Initial support for additive secret sharing.

### CODE REVIEW

File: syft/frameworks/torch/tensors/add.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        # automatically if needed)
+        response = AdditiveSharingTensor().set_shares(new_shares)
+
+        return response
+
+    @hook
+    def __sub__(self, *args, **kwargs):
+        """Subtracts two tensors. See .sub() forr details."""
+        return self.sub(*args, **kwargs)
+
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/add.py]: do we want mul?

File: test/torch/tensors/test_add.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import pytest
+import torch
+import torch.nn.functional as F
+import syft
+
+from syft.frameworks.torch.tensors import AdditiveSharingTensor
+
+
+class TestAdditiveSharingTensor(object):
```
Comment (robert-wagner) [inline test/torch/tensors/test_add.py]: Alexis put a huge amount of work refactoring all of our tests to use the pytest style. This file should be reformatted for consistency

File: test/torch/tensors/test_precision.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from syft.frameworks.torch.tensors import FixedPrecisionTensor
 
 
-class TestLoggingTensor(object):
+class TestFixedPrecisionTensor(object):
```
Comment (robert-wagner) [inline test/torch/tensors/test_precision.py]: same as other comment on test style

File: syft/frameworks/torch/tensors/add.py (hunk 1 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/add.py]: Yes - but not required to get federated learning working which is what i'm focused on atm

File: syft/frameworks/torch/tensors/add.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    #
+    # def manual_add(self, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/frameworks/torch/tensors/add.py]: oh - yeah i can remove those

[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/frameworks/torch/tensors/add.py]: done

**Pull request merged**